### PR TITLE
Improving build script

### PR DESCRIPTION
**Pull request by ErwanAliasr1**
This PR is about making the build process as a script with enhanced debugging when something gets wrong.

It does reports 
- the deleted & leftover files
- the remaining processes
- detect the segfault & extract debug info with addr2line

This makes devs's life easier as it helps understanding the output of the build.

NOTE: THIS PR IS STILL TARGETING MY PERSONAL BRANCH AND NEEDS AN UPDATE BEFORE THE MERGE.

### CODE REVIEW

**Comment by ErwanAliasr1** [comment]
Anyone ?

File: ceph-pull-requests/config/definitions/ceph-pull-requests.yml [hunk 1]
```diff
@@ -1,3 +1,30 @@
+- scm:
+    name: ceph-build
```
Comment (andrewschoen) [inline ceph-pull-requests/config/definitions/ceph-pull-requests.yml]: You shouldn't need to clone ceph-build. Just use ``!include-raw`` with the shell builders below.

See: https://github.com/ceph/ceph-build/blob/master/ceph-build/config/definitions/ceph-build.yml#L56

File: ceph-pull-requests/config/definitions/ceph-pull-requests.yml [hunk 2]
```diff
@@ -35,22 +62,15 @@
...
-          browser: auto
-          timeout: 20
-          skip-tag: true
-          wipe-workspace: true
+      - ceph-build
+      - ceph
 
     builders:
-      - shell: "ccache -M 5G; ccache -z; export NPROC=$(nproc); timeout 7200 ./run-make-check.sh; ccache -s"
+      - shell: "cd ceph; ../ceph-build/ceph-pull-requests/build/build build"
```
Comment (andrewschoen) [inline ceph-pull-requests/config/definitions/ceph-pull-requests.yml]: You can use ``!include_raw`` here.  See: https://github.com/ceph/ceph-build/blob/master/ceph-build/config/definitions/ceph-build.yml#L56

File: ceph-pull-requests/config/definitions/ceph-pull-requests.yml [hunk 3]
```diff
@@ -35,22 +62,15 @@
...
 
     publishers:
       - postbuildscript:
           script-only-if-succeeded: False
           script-only-if-failed: True
           builders:
-            - shell: "sudo reboot"
+            - shell: "cd ceph; ../ceph-build/ceph-pull-requests/build/build failure"
```
Comment (andrewschoen) [inline ceph-pull-requests/config/definitions/ceph-pull-requests.yml]: You can use ``!include_raw`` here as well.

**Pull request closed without merging**